core/src/main/scala/com/colisweb/jrubysnesshours/core/Schedule.scala

```diff
@@ -15,76 +16,103 @@ final case class Schedule private[core] (
 
       val numberOfDays = localStartDate.until(localEndDate, ChronoUnit.DAYS)
 
       val dayRangeIntervals: ListBuffer[TimeIntervalForDate] =
         (1L until numberOfDays)
           .foldLeft(ListBuffer.empty[TimeIntervalForDate]) { (acc, i) =>
             val date = localStartDate.plusDays(i)
-            acc ++ allIntervalsInDay(date)
+            acc ++ allIntervalsInDate(date)
           }
 
       startDayIntervals ++ dayRangeIntervals ++ endDayIntervals
     }
   }
 
+  def within(start: ZonedDateTime, end: ZonedDateTime): Duration =
+    intervalsBetween(start, end).map(_.duration).reduce(_ plus _)
+
   // TODO: To Test
-  def contains(date: ZonedDateTime): Boolean = {
-    val time      = date.withZoneSameInstant(timeZone).toLocalTime
-    val localDate = date.toLocalDate
+  def contains(instant: ZonedDateTime): Boolean = {
+    val localInstant = instant.withZoneSameInstant(timeZone).toLocalDateTime
+    val date         = localInstant.toLocalDate
+    val time         = localInstant.toLocalTime
 
     @inline def existsPlanning =
-      planningFor(date.getDayOfWeek).exists(_.contains(time))
+      planningFor(instant.getDayOfWeek).exists(_.contains(time))
 
     @inline def notExistsException =
-      !exceptionFor(localDate).exists(_.contains(time))
+      !exceptionFor(date).exists(_.contains(time))
 
     existsPlanning && notExistsException
   }
 
-  private[core] def intervalsInStartDay(start: ZonedDateTime): List[TimeIntervalForDate] =
-    allIntervalsInDay(start.toLocalDate, List(TimeInterval(start = LocalTime.MIDNIGHT, end = start.toLocalTime)))
+  // TODO: To Test
+  def isOpenForDurationInDate(date: LocalDate, duration: Duration): Boolean = {
+    val start = ZonedDateTime.of(date, LocalTime.MIN, timeZone)
+    val end   = ZonedDateTime.of(date, TimeInterval.END_OF_DAY, timeZone)
+
+    intervalsBetween(start, end).exists(_.duration >= duration)
+  }
+
+  def nextOpenTimeAfter(instant: ZonedDateTime): Option[ZonedDateTime] = {
+    if (planning.nonEmpty) {
 
-  private[core] def intervalsInEndDay(end: ZonedDateTime): List[TimeIntervalForDate] =
-    allIntervalsInDay(end.toLocalDate, List(TimeInterval(start = end.toLocalTime, end = TimeInterval.END_OF_DAY)))
+      @tailrec
+      def findNextOpenTimeAfter(date: LocalDate, exception: List[TimeInterval]): Option[LocalDateTime] = {
+        allIntervalsInDate(date, exception) match {
+          case interval :: _ => Some(LocalDateTime.of(date, interval.start))
+          case Nil           => findNextOpenTimeAfter(date.plusDays(1), Nil)
+        }
+      }
+      val localInstant = instant.withZoneSameInstant(timeZone).toLocalDateTime
+      val date         = localInstant.toLocalDate
+      val time         = localInstant.toLocalTime
 
-  private[core] def intervalsInSameDay(
-      date: LocalDate,
-      query: TimeInterval
-  ): List[TimeIntervalForDate] =
-    allIntervalsInDay(
+      findNextOpenTimeAfter(date, List(TimeInterval(LocalTime.MIDNIGHT, time)))
```

Comment:
Replace `LocalTime.MIDNIGHT` by `LocalTime.MIN` plz.

Can you also do it everywhere, plz ?